### PR TITLE
Add support for getting the UUID of a device

### PR DESCRIPTION
**Pull request by vmx**
`Device::uuid()` was added to get the UUID of a device via the underlying
`cuDevuceGetUuid()` call.

### CODE REVIEW

File: crates/cust/src/device.rs [hunk 1]
```diff
@@ -327,6 +327,29 @@ impl Device {
...
+    /// let device = Device::get_device(0)?;
+    /// println!("Device UUID: {:?}", device.uuid()?);
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn uuid(self) -> CudaResult<[u8; 16]> {
+        unsafe {
+            let mut cu_uuid = CUuuid { bytes: [0i8; 16] };
+            cuDeviceGetUuid(&mut cu_uuid, self.device).to_result()?;
+            let uuid: [u8; 16] = ::std::mem::transmute(cu_uuid.bytes);
```
Comment (RDambrosio016) [inline crates/cust/src/device.rs:348]: Using a transmute for just `i8` -> `u8` seems a bit odd, just using `array.map()` would be better.

Comment (vmx) [reply crates/cust/src/device.rs]: Fixed with the most recent push (I also fixed the commit message to match the usual style).

File: crates/cust/src/device.rs [hunk 2]
```diff
@@ -327,6 +327,29 @@ impl Device {
...
+    /// println!("Device UUID: {:?}", device.uuid()?);
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn uuid(self) -> CudaResult<[u8; 16]> {
+        let mut cu_uuid = CUuuid { bytes: [0i8; 16] };
+        unsafe {
+            cuDeviceGetUuid(&mut cu_uuid, self.device).to_result()?;
+        }
+        let uuid: [u8; 16] = cu_uuid.bytes.map(|byte| byte.to_ne_bytes()[0]);
```
Comment (RDambrosio016) [inline crates/cust/src/device.rs:349]:
```suggestion
        let uuid: [u8; 16] = cu_uuid.bytes.map(|byte| byte as u8);
```

**Comment by vmx** [comment]
I can also squash and rebase the commit if you prefer that.

**Comment by RDambrosio016** [comment]
I usually squash commits anyways, so this is fine :)

**Pull request merged**